tests/nightly/gpu/test_torchscript.py

```diff
@@ -62,6 +62,72 @@ def test_token_splitter(self):
                     if idx + 1 == num_examples:
                         break
 
+    def test_special_tokenization(self):
+        from parlai.core.dict import DictionaryAgent
+        from parlai.core.params import ParlaiParser
+        from parlai.torchscript.modules import ScriptableDictionaryAgent
+
+        SPECIAL = ['Q00', 'Q01']
+        text = "Don't have a Q00, man! Have a Q01 instead."
+
+        parser = ParlaiParser(False, False)
+        DictionaryAgent.add_cmdline_args(parser)
+        with testing_utils.tempdir() as tmp:
+            opt = parser.parse_kwargs(
+                dict_tokenizer='gpt2', dict_file=os.path.join(tmp, 'dict')
+            )
+
+            orig_dict = DictionaryAgent(opt)
+
+            orig_bpe = orig_dict.bpe
+            fused_key_bpe_ranks = {
+                '\n'.join(key): float(val) for key, val in orig_bpe.bpe_ranks.items()
+            }
+
+            sda = ScriptableDictionaryAgent(
+                null_token=orig_dict.null_token,
+                end_token=orig_dict.end_token,
+                unk_token=orig_dict.unk_token,
+                start_token=orig_dict.start_token,
+                freq=orig_dict.freq,
+                tok2ind=orig_dict.tok2ind,
+                ind2tok=orig_dict.ind2tok,
+                bpe_add_prefix_space=False,
+                bpe_encoder=orig_bpe.encoder,
+                bpe_byte_encoder=orig_bpe.byte_encoder,
+                fused_key_bpe_ranks=fused_key_bpe_ranks,
+                special_tokens=[],
+            )
+
+            tokenized = sda.txt2vec(text)
+            assert len(tokenized) == 15
+            assert sda.vec2txt(tokenized) == text
+            nice_tok = [sda.ind2tok[i] for i in tokenized]
+
+            orig_dict = DictionaryAgent(opt)
+            orig_dict.add_additional_special_tokens(SPECIAL)
+            orig_bpe = orig_dict.bpe
+            sda = ScriptableDictionaryAgent(
+                null_token=orig_dict.null_token,
+                end_token=orig_dict.end_token,
+                unk_token=orig_dict.unk_token,
+                start_token=orig_dict.start_token,
+                freq=orig_dict.freq,
+                tok2ind=orig_dict.tok2ind,
+                ind2tok=orig_dict.ind2tok,
+                bpe_add_prefix_space=False,
+                bpe_encoder=orig_bpe.encoder,
+                bpe_byte_encoder=orig_bpe.byte_encoder,
+                fused_key_bpe_ranks=fused_key_bpe_ranks,
+                special_tokens=SPECIAL,
+            )
+
+            special_tokenized = sda.txt2vec(text)
+            assert len(special_tokenized) == 15
+            assert sda.vec2txt(special_tokenized) == text
+            assert special_tokenized != tokenized
```

Comment:
Thought: it could be even more explicit to check the actual strings of the output tokens, instead of just their length and whether they match with/without special tokens. No strong opinion on this either way, though